File: an2_sem4/MPP/Homework/lab2/java/repo/src/main/java/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/repo/IRepo.java
package java.ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Entity;

import java.ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.exceptions.EntityRepoException;

public interface IRepo <ID, E extends Entity<ID>> {
    E add(E e) throws EntityRepoException;
    E update(E e) throws EntityRepoException;
    E remove(ID id) throws EntityRepoException;
    E getById(ID id) throws EntityRepoException;
    Iterable<E> getAll()throws EntityRepoException;
}

File: an2_sem4/ISS/ISS.Biblioteca/ISS.Biblioteca.Commons/Repo/IRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ISS.Biblioteca.Commons.Repo
{
    public interface IRepo<E>
    {
        E Add(E entity);
        E Update(E entity);
        E Remove(E entity);
        IEnumerable<E> GetAll();
        E GetById(int id);       
    }
}
